lib/sqlalchemy/orm/dynamic.py

```diff
@@ -21,35 +20,56 @@
 from __future__ import annotations
 
 from typing import Any
+from typing import Callable
 from typing import Iterable
 from typing import Iterator
+from typing import Optional
+from typing import Tuple
+from typing import Type
 from typing import TYPE_CHECKING
 from typing import TypeVar
-
+from typing import Union
+
+from sqlalchemy.engine.result import Result
+from sqlalchemy.engine.result import ScalarResult
+from sqlalchemy.orm.mapper import Mapper
+from sqlalchemy.orm.query import Query
+from sqlalchemy.orm.state import InstanceState
+from sqlalchemy.util.langhelpers import symbol
 from . import attributes
 from . import exc as orm_exc
+from . import QueryableAttribute
 from . import relationships
 from . import util as orm_util
-from .query import Query
 from .session import object_session
+from .util import AliasedClass
 from .writeonly import AbstractCollectionWriter
 from .writeonly import WriteOnlyAttributeImpl
+from .writeonly import WriteOnlyCollection
 from .writeonly import WriteOnlyHistory
 from .writeonly import WriteOnlyLoader
 from .. import util
 from ..engine import result
+from ..event import _Dispatch
```

Comment:
@mlatysh 

All internal modules must be imported relatively and put under an `if TYPE_CHECKING` branch if not used at runtime.

Example: https://github.com/GodelTech/sqlalchemy/pull/3/files#diff-a178567dd254a0b999fc247ba30743808c234d94f43b7cc88d878be1ec31fe55R40-R55

Reply:
@g-kisenkov @dzmitar Refactored as mentioned, made new imports under `if TYPE_CHECKING` condition and made all of them relative